samza-api/src/main/java/org/apache/samza/application/StreamApplication.java

```diff
@@ -16,17 +16,20 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-package org.apache.samza.operators;
+package org.apache.samza.application;
 
 import org.apache.samza.annotation.InterfaceStability;
 import org.apache.samza.config.Config;
+import org.apache.samza.operators.StreamGraph;
 
 
 /**
  * This interface defines a factory class that user will implement to create user-defined operator DAG in a {@link StreamGraph} object.
  */
 @InterfaceStability.Unstable
-public interface StreamGraphBuilder {
+public interface StreamApplication {
+  static final String APP_CLASS_CONFIG = "app.class";
```

Comment:
I have another patch working on adding AppConfig to wrap those configure variables as in other configs. Neither here nor the TaskFactoryUtils is the right place. I simply placed it here to be closer to its own usage, for the short term. Planning to move it to AppConfig.